install/kubernetes/helm/istio/charts/gateways/templates/deployment.yaml

```diff
@@ -153,20 +151,12 @@ spec:
           - --controlPlaneAuthPolicy
           - MUTUAL_TLS
           - --discoveryAddress
-          {{- if $.Values.global.istioNamespace }}
-          - istio-pilot.{{ $.Values.global.istioNamespace }}:15011
-          {{- else }}
-          - istio-pilot:15011
-          {{- end }}
+          - istio-pilot.{{ $.Release.Namespace }}:15011
```

Comment:
I thought this was intended so you `helm template istio --namespace istio-system` then later `helm template -x gateways --namespace gateway-ns --set global.istioNamespace=istio-system`?

So release namespace doesn't always refer to the control plane in these cases

Reply:
Correct, this was intended.